### DISCUSSION TITLE
Release 1.2.0 Beta 10

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#967] [General]
## GENERAL
- **NEW** Multi-Mono support (resolves #955)
- Generate a unique server ID to allow for auto reconnects with IP/name changes - This improves session stability even when servers change their IP address.
- Improved server error reporting in the plugin

## PLUGIN
- **NEW** Allow plugin layouts to be selected
- **NEW** Added option to use fixed output buffers
- **NEW** Added tray option to reload all chains in "not loaded" state
- **NEW** Added option to manually adjust the latency (PDC)
- **NEW** Keep transferring audio when a plugin UI is open
- Changed FX channel limits to 32 inputs + 2 sidechains
- Fixed audio streamer issue with uneven buffer sizes (resolves #948)
- Fixed AAX installation on windows
- Fixed layout issue with the instrument plugin that prevented stereo outputs in Logic
- Stability fixes

## SERVER
- Improved plugin I/O management (layouts cache)
- Delete plugin UIs only when the plugin unloads
- Fixed plugin deactivation support (resolves #947)
- Fixed performance issue in the processing measurement code
- Stability fixes

All downloads are available at https://audiogridder.com/download/.

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta10'>Release 1.2.0 Beta 10</a>.</em>

**Comment by Pabloscotty** [comment]
Thank you Anderas. But this version crashes upon loading cache and plugins. Windows 7 - Server and Mac 10.14 Plugin

**Reply by Analog4Lyfe** [reply]
Do a wipe and rescan

**Reply by apohl79** [reply]
Can you try to reset the server and force it to do clean scan again? You can find the location here C:\Users\<User>\AppData\Roaming\AudioGridder\.

**Reply by Pabloscotty** [reply]
Nothing change after resetting. It won't load some of the plugins at scan (which works prefectly in BETA 8). Crashes and won't load plugin GUI

![Screenshot 2022-07-07 at 09 41 24](https://user-images.githubusercontent.com/80844841/177721784-203039f8-15b6-4864-b135-69a8b43018c3.jpg)
![Screenshot 2022-07-07 at 09 41 45](https://user-images.githubusercontent.com/80844841/177721806-5dce8e7b-bec7-45c0-9f1e-869e03d8d314.jpg)
![Screenshot 2022-07-07 at 09 42 23](https://user-images.githubusercontent.com/80844841/177721813-59f1c6ee-6d63-4887-9ddb-a82238653190.jpg)
![Screenshot 2022-07-07 at 09 42 51](https://user-images.githubusercontent.com/80844841/177721818-c819d39a-abff-48ca-8f98-20e5e17ca414.jpg)
![Screenshot 2022-07-07 at 09 46 17](https://user-images.githubusercontent.com/80844841/177721820-97627d11-95d2-40d0-af64-596191c0e1b9.jpg)

This is from BETA 8

![Screenshot 2022-07-07 at 09 54 05](https://user-images.githubusercontent.com/80844841/177721826-a8d8531a-f975-4a2a-ae5d-72d7c7d60a05.jpg)

**Reply by apohl79** [reply]
Can you try to remove the red ones from the blacklist in the plugin manager and click rescan? 

if that does not help, could you provide the scan logs? cheers!

**Reply by Pabloscotty** [reply]
[Scan.zip](https://github.com/apohl79/audiogridder/files/9061681/Scan.zip)

**Reply by apohl79** [reply]
Thx! In the latest scan you did, was weiss eq and kontakt still failing? The logs at least say they were successful. So it might actually be a bug somewhere else.

**Comment by emack999** [comment]
Lots of issues with plugin scanning.  I've done a wipe and rescan, but many plugins are being prompted to wait or abort numerous times, including all BlueCat and Waves plugins.  The Waves shells took about 6-7 tries to get a complete scan. These plugins weren't having issues in previous versions, so I'm guessing the timeout period was shortened?  In any case, its a much longer process for the initial scan now., and many more failed plugin that required removing from the blacklist and rescanning.

Also noticing some display issues using Arturia FX plugins, where the preset menu won't stay open---it opens for a few milliseconds and closes on its own.

**Reply by Pete3030** [reply]
I have a problem with the Beta10 plugin, in Logic my old session won't open it, giving me the message "Das Plug-In „AGridder“  ist auf deinem System nicht verfügbar.". I had Beta 6 installed before that, maybe this is worth mentioning. AU validation of the Beta10 plugin is fine though. And: In Reaper the Beta10 plugin loads fine within an old session, so it seems like this is a Logic specific problem? I am on Logic 10.4.8 btw. Any suggestions would be greatly appreciated..!

**Reply by apohl79** [reply]
> Lots of issues with plugin scanning.  I've done a wipe and rescan, but many plugins are being prompted to wait or abort numerous times, including all BlueCat and Waves plugins.  The Waves shells took about 6-7 tries to get a complete scan. These plugins weren't having issues in previous versions, so I'm guessing the timeout period was shortened?  In any case, its a much longer process for the initial scan now., and many more failed plugin that required removing from the blacklist and rescanning.

this is unfortunate. the scanner is testing layouts in the new version, which makes it a bit slower and unstable as it seems.

if you do not get all your plugins scanned successfully, you can revert to a prev version, do a full rescan and upgrade back to beta10. 

> Also noticing some display issues using Arturia FX plugins, where the preset menu won't stay open---it opens for a few milliseconds and closes on its own.

not sure what is causing this. will have to wait and see if more problems like this get reported.

**Reply by apohl79** [reply]
> I have a problem with the Beta10 plugin, in Logic my old session won't open it, giving me the message "Das Plug-In „AGridder“ ist auf deinem System nicht verfügbar.". I had Beta 6 installed before that, maybe this is worth mentioning. AU validation of the Beta10 plugin is fine though. And: In Reaper the Beta10 plugin loads fine within an old session, so it seems like this is a Logic specific problem? I am on Logic 10.4.8 btw. Any suggestions would be greatly appreciated..!

can you
1) do a "full audio unit reset" in logics plugin manager
2) close logic
3) run `rm -rf ~/Library/Caches/AudioUnitCache/` in a terminal app
4) reboot
5) start logic (which does a full rescan of all plugins)

I hope this helps.

**Reply by musicgreator** [reply]
I just tested Beta10 with a session created with a earlyer version. After playing it back one two minutes, one of my AG instances repeatedly disconnects (after reopening the session and trying again). Once i double clicked the failing instance in the AG Monitor and it crashed Reaper. The failing instance is running one Unisum Compressor. 

Using Windows 10 and Reaper, running AG locally.

**Comment by Pete3030** [comment]
Hi Andreas, I tried it twice, but unfortunately the Plugin is still not recognized by Logic.. Is there maybe anything else I could try? Otherwise I guess I have to revert to Beta7 again..

**Reply by Centripidity** [reply]
Have you tried rebooting your Mac?  Several times I have found Logic doesn’t recognize new plugins until a restart.

**Reply by Pete3030** [reply]
Well i did a restart several times and plugin is recognized by Logic if I open Audiogridder as a new instance (in my old project). But: The old instances auf Audiogridder (in my old project) appear with the yellow ! symbol, and Logic tells me that for those instances the plugin is not installed.. My guess is that the only fix for that can come from Andreas (in Beta11 hopefully)..

**Reply by Centripidity** [reply]
It's sounds like there is some change between beta 9 and 10 that makes 10 unable to either recognise it as the same plugin or to configure itself the way you have it in your project.  Andreas would be the best person to look at that.  The AU plugin ID hasn't changed between the two versions but there has been a change in I/O.  It has gone from 16 in/16 out to 30 in/32 out.  Perhaps it can't find the I/O configuration you have chosen in the new version.

Of course, I'm only guessing.

**Reply by Pete3030** [reply]
Good point, I am not an expert, but this could be the potential root cause for it.. As I said, in  Reaper I don't have this problem, maybe Logic has it's specifics when it comes to Plugin recognition..;)

**Reply by Centripidity** [reply]
Also, are you using the AU in Logic and the VST in Reaper?

I love Logic but it has some serious, and sometimes annoying idiosyncrasies.

**Reply by Pete3030** [reply]
No, I also use the AU version in Reaper..

**Reply by PierreMConsulting** [reply]
Same issue here with Studio One Pro. Audiogridder plugin (previously beta 9) is not recognized as installed anymore. Also had a lot of plugin scan issues on both my MacOS (Catalina) and Windows (10) servers. Reverted to Beta 9

**Comment by mixmonkey** [comment]
Using beta10 on macOS Monterey 12.3.1/Reaper 6.63. A few observations, using the TAL-Drum instrument (VST2):
i) AG doesn't seem to automatically compensate for the delay buffer it uses (pretty sure it used to). I always have to set a manual delay equal to the buffer size. This may be intentional?
ii) If I offline the AG plugin in Reaper, when I bring it back online I have to restart the server before it will work.
iii) Very occasionally, after the server restarts, playback will be out of sync by half the actual buffer size. Restarting again usually cures it, but it's a bit of a craps shoot.
iv) Issues with plugin UI elements not working, but only in Local Mode.

**Comment by fojerhar** [comment]
Hi Andreas
Maybe you know that, beta10 is not downward compatible. I had to open a session with beta8 and all the slots were empty, none of the plugins were present in beta8.
thank you for your hard work

**Reply by PierreMConsulting** [reply]
Had the same issue it seems it's the case for AU (I did not check with other formats as I don't have any previous session with vst or vst3). I manually removed the plugin files and reverted back to beta 9 for plugin & server. In case you need to do the same Andreas made it easy for us to do so just go to https://audiogridder.com/releases/ where you'll find all the releases

**Comment by emerymastering** [comment]
a very nice release, my setup likes it:) All works! Big ups!

**Comment by apohl79** [comment]
> Using beta10 on macOS Monterey 12.3.1/Reaper 6.63. A few observations, using the TAL-Drum instrument (VST2):
> 
> i) AG doesn't seem to automatically compensate for the delay buffer it uses (pretty sure it used to). I always have to set a manual delay equal to the buffer size. This may be intentional?

no does not sound right. i don't want to assume i did not break anything, but i'm sure the behavior should not be different in b10 than in previous versions.

what plugin format of AG are you using in reaper? is it the the same with other formats? could you collect the server and plugin logs and provide them?

> ii) If I offline the AG plugin in Reaper, when I bring it back online I have to restart the server before it will work.

should not be needed. maybe the tal plugin is causing some unintended side effect in the server. the logs might help explaining what's going on.

> iii) Very occasionally, after the server restarts, playback will be out of sync by half the actual buffer size. Restarting again usually cures it, but it's a bit of a craps shoot.

seems related to 1.

> iv) Issues with plugin UI elements not working, but only in Local Mode.

is this new in b10?

**Reply by mixmonkey** [reply]
I repeated all the tests with a different sampler VST (Sitala) to rule out TAL-Drum being responsible. Results were the same.

i) Definitely different behaviour to beta9. Beta9 automatically compensates for the buffer set. Confirmed by re-installing beta9. Happens with both VST2 and VST3 in beta10. Logs attached.
[AudioGridder.zip](https://github.com/apohl79/audiogridder/files/9079998/AudioGridder.zip)

ii) Further investigation reveals that the Instrument Outputs/Enable all channels is not re-enabled when the plugin is brought back online. Enabling all outputs manually solves the issue.

iii) In beta9 sync is always perfect after plugin is brought back online.

iv) All the UI issues are caused by using Reaper in full screen mode. None occur when running in a window on the desktop. This happens in beta9 also.

**Reply by apohl79** [reply]
thx for the details!

**Reply by mixmonkey** [reply]
Sync/buffer issues in beta10 only affect the Instrument version of the AG plugin. Pure audio version works as expected.

Instrument outputs not being re-enabled when plugin brought back online also happens on project save/load.

**Comment by apohl79** [comment]
> Hi Andreas
> 
> Maybe you know that, beta10 is not downward compatible. I had to open a session with beta8 and all the slots were empty, none of the plugins were present in beta8.

do you mean, that you do not find the plugins previously available (in b8) in the plugin search in b10? if yes, please disable the server filter in the plugin settings menu. i'm considering to remove the filter entirely. it seems that it got more restrictive with the layout/multi-mono changes. so i think it should go away.

> thank you for your hard work

thanks! 🙏

**Reply by fojerhar** [reply]
Hi
Nope. I was in the middle of a mixing session, and i updated to beta10, and continued mixing with it. After a while I had some issues (beachball freezing in protools-my computers fault not yours!), and re installed beta8 what works for me well.
Reopened the session, and all the AG plugins were empty, so I went back to my last backup what was made with beta8.

**Reply by apohl79** [reply]
ok, thats not good. I do backwards compatibility tests, but did not see this happening. Thx for letting me know. You should always be able to switch back to an old version w/o loosing plugin settings.

**Comment by apohl79** [comment]
> a very nice release, my setup likes it:) All works! Big ups!

cheers man, glad to hear ☺️

**Comment by musicgreator** [comment]
I just tested Beta10 with a session created with a earlyer version. After playing it back one two minutes, one of my AG instances repeatedly disconnects (after reopening the session and trying again). Once i double clicked the failing instance in the AG Monitor and it crashed Reaper. The failing instance is running one Unisum Compressor.

After i re-installed Beat6, i can`t open that session anymore, crashes while loading. The session was created using Beta6.

Using Windows 10 and Reaper, running AG locally.

**Reply by apohl79** [reply]
Ok, looks like the backwards compatibility is broken in b10. needs further investigation and a fix.

**Comment by djniviro** [comment]
Looks like an awesome update! Any guidance on how to enable the multi-mono feature and where to find its settings? I might have overlooked something here.

**Reply by apohl79** [reply]
The server has to know the plugin layouts (it complains if it does not). Whenever you put AG on a multi-channel insert now, you will see the multi-mono layout being available for every plugin, that supports mono.

**Comment by rickvdk** [comment]
Hi! I'm on Windows 10 Ryzen 3900x - at first this release seemed to work great. fast load times, less crashing etc. was really happy but then it started loading Acustica vst FX in a phasey Mono signal when loading as a stereo plugin. It did not do this before. I've tried re-installing, cleaning cache etc. to no avail. What could be the issue? Thanks for your efforts creating this amazing CPU saver!!!

**Comment by Henri75** [comment]
Just back to AG and was happy to see the b10 :) lot of things seems better 

I can confirm weird scanning issues with beta 10.

1) Tons of plugins (got 1500+) now got tons of timeout error during scanning ! Way more than before !
In fact on 2 laptops I am still trying to scan everything for 2 days ... after a tons of timeout error AG crashes and restart ... from the beginning.

So it seems there is an issue with the scanning phase. Maybe a longer timeout  value is needed ? Or a value we can set in the config ?  Or set some auto retry for X times ? 
Because at the end most of timeout will passes after 2 to 4 retry (except some waves and medla).

2) At the beginning I didnt wanted to rescan everything, but : 
- on some computers AG is stuck on the new message 'plugin cache' and nothing happens
-  it seems that a rescan is needed anyway because of the new IO layout map.

**Comment by CaBuffalo** [comment]
Clear the cache and do a full recan should fix this issue.

On Fri, Jul 15, 2022 at 8:00 AM Henri75 ***@***.***> wrote:

> Just back to AG and was happy to see the b10 :) lot of things seems better
>
> I can confirm weird scanning issues with beta 10.
>
>    1. Tons of plugins (got 1500+) now got tons of timeout error during
>    scanning ! Way more than before !
>    In fact on 2 laptops I am still trying to scan everything for 2 days
>    ... after a tons of timeout error AG crashes and restart ... from the
>    beginning.
>
> So it seems there is an issue with the scanning phase. Maybe a longer
> timeout value is needed ? Or a value we can set in the config ? Or set some
> auto retry for X times ?
> Because at the end most of timeout will passes after 2 to 4 retry (except
> some waves and medla).
>
>    1. At the beginning I didnt wanted to rescan everything, but :
>
>
>    - on some computers AG is stuck on the new message 'plugin cache' and
>    nothing happens
>    - it seems that a rescan is needed anyway because of the new IO layout
>    map.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/967#discussioncomment-3157527>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6AFOAISMXYNG3ZDN4TVUF4HPANCNFSM522SOJ6A>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by PierreMConsulting** [reply]
Did not fix it for me on both MacOS & Windows 10

**Comment by Henri75** [comment]
Doesnt fix it. I tried that first of course.

Now on some computers I am stuck being unable to rescan up to completion
all the plugins with beta 10 !
Works with previous versions.

Le ven. 15 juil. 2022 à 17:11, CaBuffalo ***@***.***> a
écrit :

> Clear the cache and do a full recan should fix this issue.
>

**Comment by Henri75** [comment]
i noticed too that after any crashes in AG during the rescan, it restarts from zero !

it should restart from the last working scan state. That is how all DAW works from what I saw, like Cubase or S1.
Moreover any plugin that crashes the DAW is blacklisted, so It wont be rescan.
So if you have an issue with one plugin, your daw still start.

Right now for 2 days I am unable to scan two computers. And after couple of hours to scan and 100+ timeout warnings I got a crash, and then when AG restart it rescan from zero !

Basically I am stuck in a loop and unable to run AG (worked fine when scanned with previous version)

**Comment by naokiapri** [comment]
for some reason most of plugins are now have only multi-mono layout and the settings are not linked, so i have to edit both L and R separately. Also it makes stereo-link settings in plugins unusable. 

though some of the plugins are with default layout and are working great, including surround ones. Is there any way to edit the layouts available to plugin?

**Comment by aryschien** [comment]
Hi Andreas,

I got mono result from 1.2.0 Beta 10, be it universal, intel or apple silicon, Pro Tools 2022.6.  I tried AG on my old Mac (OS 10.15.7) and the new Mac Studio (OS 12.4), launching a stereo instance of AG plug-in, and all got mono result from a stereo audio file.

It was fine with 1.2.0 Beta 6, stereo in stereo out.

Thanks in advance.

**Comment by fuzzfaze** [comment]
Hi
Same result here / mono instances on protools ultimate 2022.4 Mac mini M1/Monterey / Beta 9 working ok
Thanks

**Comment by Ch33ftain** [comment]
Probably a stupid question but can I run AAX plugins in Ableton with AG?

**Reply by apohl79** [reply]
No, AG supports hosting VST2, VST3 and AU currently.

**Comment by orkunsunor** [comment]
hello ı use logic pro x mac ox montery logic pro x version 10.7.9 ı install Beta 9 but not show  logic pro au ı dint understood please help

**Comment by orkunsunor** [comment]
anyone have crossgridder 1.5 or 1.4 oe where ı find it please help guys

**Comment by orkunsunor** [comment]
Which version of Audio Gridder Sequioa 15.7.2 do I need to install because Logic Pro X doesn't recognize it?
